packages/block-library/src/navigation/style.scss

```diff
@@ -327,107 +314,29 @@
 	.is-responsive {
 		display: none;
 	}
-
-	// Horizontal layout
-	display: flex;
-	flex-wrap: wrap;
-
-	.items-justified-space-between & {
-		flex-grow: 1;
-	}
-}
-
-// Vertical layout
-.is-vertical .wp-block-page-list, // Page list.
-.is-vertical .wp-block-navigation__container,
-.is-vertical .wp-block-navigation__responsive-container-content {
-	display: flex;
-	flex-direction: column;
-	align-items: flex-start;
-}
-
-// Justification.
-// These target the named container class to work even with the additional mobile menu containers.
-.items-justified-center {
-	.wp-block-page-list,
```

Comment:
These rules might be ones we need to restore, at least the responsive container parts as they affect the contents of the overlay.

Essentially, if you justify your navigation block to be centered, so should the content of the overlay be centered. The tricky piece comes with the dual variations, horizontal and vertical for the block itself, regardless of that orientation, the content of the overlay should always be vertically aligned to the top, so we can't just inherit the rules, and we have to have separate rules for the vertical and horizontal orientations.

Reply:
Justification inside the overlay should be working now. In any case, these styles no longer apply when using layout because `items-justified-center` etc. classnames aren't added to the markup.

> The tricky piece comes with the dual variations

😅 yup, it was tricky all right. In order to make it work, I added a custom property `--justification-setting` to the layout-generated styles that get added to the block container. That property is then used to set justification inside the overlay regardless of whether the block is oriented vertically or horizontally. In my local it works well and content is always vertically aligned to the top, but as I mentioned above you may need to pull down this branch again to test the latest version.